art/experiment/Experiment.py

```diff
@@ -24,43 +24,61 @@ def __init__(self, name, **kwargs):
 
     def add_step(self, step: "Step", checks: List[Check]):
         self.steps.append(step)
-        self.state.step_states[step.get_model_name()][step.get_name_with_id()] = step._get_saved_state()
         step.set_step_id(len(self.steps))
         self.checks.append(checks)
 
+    def fill_step_states(self, step):
+        self.state.step_states[step.get_model_name()][
+            step.get_name_with_id()
+        ] = step.get_results()
+
     def run_all(self):
```

Comment:
The whole run_all function has grown a lot. Should we consider using more functions to keep it clear?

Just loudly thinking

Reply:
You are right. I did small refactor